File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskCreateTask.java
package net.imatruck.dsmanager.tasks;

import android.app.Activity;
import android.os.AsyncTask;
import android.widget.TextView;

import net.imatruck.dsmanager.R;
import net.imatruck.dsmanager.models.DSTaskCreateBase;
import net.imatruck.dsmanager.utils.SynologyDSTaskError;

import java.io.IOException;

import retrofit2.Call;

/**
 * Created by marc on 2017-03-24.
 */

public class DSTaskCreateTask extends AsyncTask<Call<DSTaskCreateBase>, Void, DSTaskCreateBase> {

    private Activity context;

    public DSTaskCreateTask(Activity context) {
        this.context = context;
    }

    @SafeVarargs
    @Override
    protected final DSTaskCreateBase doInBackground(Call<DSTaskCreateBase>... calls) {

        DSTaskCreateBase dsTaskListBase;
        try {
            dsTaskListBase = calls[0].execute().body();
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }

        return dsTaskListBase;
    }

    @Override
    protected void onPostExecute(DSTaskCreateBase dsInfo) {
        if (dsInfo != null) {
            boolean success = dsInfo.isSuccess();
            TextView debugTextView = (TextView) context.findViewById(R.id.debug_api_text);

            if (success) {
                String text = "Task Create: Success";
                debugTextView.setText(text);
            } else {
                String text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.getError().getCode()));
                debugTextView.setText(text);
            }
        }
    }
}

File: app/src/main/java/net/imatruck/dsmanager/tasks/DSTaskCreateTask.kt
package net.imatruck.dsmanager.tasks

import android.app.Activity
import android.os.AsyncTask
import android.widget.TextView
import net.imatruck.dsmanager.R
import net.imatruck.dsmanager.models.DSTaskCreateBase
import net.imatruck.dsmanager.utils.SynologyDSTaskError
import retrofit2.Call
import java.io.IOException


class DSTaskCreateTask(private val context: Activity) : AsyncTask<Call<DSTaskCreateBase>, Void, DSTaskCreateBase>() {

    @SafeVarargs
    override fun doInBackground(vararg calls: Call<DSTaskCreateBase>): DSTaskCreateBase? {

        val dsTaskListBase: DSTaskCreateBase
        try {
            dsTaskListBase = calls[0].execute().body()
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }

        return dsTaskListBase
    }

    override fun onPostExecute(dsInfo: DSTaskCreateBase?) {
        if (dsInfo != null) {
            val success = dsInfo.isSuccess
            val debugTextView = context.findViewById<TextView>(R.id.debug_api_text)

            if (success) {
                val text = "Task Create: Success"
                debugTextView.text = text
            } else {
                val text = context.getString(
                        SynologyDSTaskError.getMessageId(dsInfo.error.code))
                debugTextView.text = text
            }
        }
    }
}
